geopmpy/geopmpy/io.py

```diff
@@ -890,13 +894,13 @@ def load_reports(self, reports, dir_name, dir_cache, verbose, do_cache):
                     try:
                         if verbose:
                             sys.stdout.write('Generating HDF5 files... ')
-                        try:
-                            self._reports_df.to_hdf(self._report_h5_name, 'report', format='table')
-                        except AttributeError:
-                            pass
-                        self._app_reports_df.to_hdf(self._report_h5_name, 'app_report', format='table', append=True)
-                        self._unmarked_reports_df.to_hdf(self._report_h5_name, 'unmarked_report', format='table', append=True)
-                        self._epoch_reports_df.to_hdf(self._report_h5_name, 'epoch_report', format='table', append=True)
+                        self._app_reports_df.to_hdf(self._report_h5_name, 'app_report', format='table')
+                        if len(self._reports_df) > 0:
+                            self._reports_df.to_hdf(self._report_h5_name, 'report', format='table', append=True)
+                        if len(self._unmarked_reports_df) > 0:
+                            self._unmarked_reports_df.to_hdf(self._report_h5_name, 'unmarked_report', format='table', append=True)
+                        if len(self._epoch_reports_df) > 0:
+                            self._epoch_reports_df.to_hdf(self._report_h5_name, 'epoch_report', format='table', append=True)
```

Comment:
Why are the `if(len)` checks needed now (or were they needed all along)?

It seems like it has something to do with the fact that we initialize these through multiple maybe-executed lines because of a string-related issue.